bundles/org.openhab.binding.iaqualink/src/main/java/org/openhab/binding/iaqualink/internal/handler/IAqualinkHandler.java

```diff
@@ -0,0 +1,450 @@
+import org.eclipse.smarthome.core.library.types.StringType;
+import org.eclipse.smarthome.core.thing.Channel;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.thing.binding.builder.ChannelBuilder;
+import org.eclipse.smarthome.core.thing.binding.builder.ThingBuilder;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.RefreshType;
+import org.eclipse.smarthome.core.types.State;
+import org.eclipse.smarthome.core.types.UnDefType;
+import org.openhab.binding.iaqualink.internal.IAqualinkBindingConstants;
+import org.openhab.binding.iaqualink.internal.api.IAqualinkClient;
+import org.openhab.binding.iaqualink.internal.api.IAqualinkClient.NotAuthorizedException;
+import org.openhab.binding.iaqualink.internal.api.model.AccountInfo;
+import org.openhab.binding.iaqualink.internal.api.model.Auxiliary;
+import org.openhab.binding.iaqualink.internal.api.model.Device;
+import org.openhab.binding.iaqualink.internal.api.model.OneTouch;
+import org.openhab.binding.iaqualink.internal.config.IAqualinkConfiguration;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.JsonArray;
+import com.google.gson.JsonElement;
+
+/**
+ *
+ * iAquaLink Control Binding
+ *
+ * iAquaLink controllers allow remote access to Jandy/Zodiac pool systems. This
+ * binding allows openHAB to both monitor and control a pool system through
+ * these controllers.
+ *
+ * The {@link IAqualinkHandler} is responsible for handling commands, which
+ * are sent to one of the channels.
+ *
+ * @author Dan Cunningham - Initial contribution
+ */
+@NonNullByDefault
+public class IAqualinkHandler extends BaseThingHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(IAqualinkHandler.class);
+
+    /**
+     * Minimum amount of time we can poll for updates
+     */
+    private static final int MIN_REFRESH_SECONDS = 5;
+
+    /**
+     * Minimum amount of time we can poll after a command
+     */
+    private static final int COMMAND_REFRESH_SECONDS = 2;
+
+    /**
+     * Default iAqulink key used by existing clients in the marketplace
+     */
+    private static final String DEFUALT_API_KEY = "EOOEMOW4YR6QNB07";
+
+    /**
+     * Local cache of iAqualink states
+     */
+    private Map<String, State> stateMap = Collections.synchronizedMap(new HashMap<>());
+
+    /**
+     * Our poll rate
+     */
+    private int refresh;
+
+    /**
+     * fixed API key provided by iAqualink clients (Android, IOS), unknown if this will change in the future.
+     */
+    @Nullable
+    private String apiKey;
+
+    /**
+     * Optional serial number of the pool controller to connect to, only useful if you have more then one controller
+     */
+    @Nullable
```

Comment:
Please place the annotation between the `private` and the type. Check all.

Reply:
I can do that.  Question, i notice there are bindings in our code base that use both styles, I'm not partial one way or the other, any benefit other then consistency or preference ?